interface/modules/custom_modules/oe-module-weno/src/Services/WenoValidate.php

```diff
@@ -0,0 +1,264 @@
+<?php
+
+namespace OpenEMR\Modules\WenoModule\Services;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+
+class WenoValidate extends ModuleService
+{
+    private string $requestUrl = 'https://online.wenoexchange.com/webapi/restapi/WenoManage';
+    private string $messageID;
+    private string $userEmail;
+    private string $md5UserPassword;
+    private mixed $encryptionKey;
+    private Client $client;
+
+    public function __construct()
+    {
+        parent::__construct();
+        $this->generateMessageID();
+        $this->setMessageProperties();
+        $this->client = new Client();
+    }
+
+    /**
+     * @return void
+     */
+    private function generateMessageID(): void
+    {
+        $random_string = substr(str_shuffle("0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"), 0, 5);
+        $timestamp = time();
+        $this->messageID = substr($timestamp . $random_string, 0, 10);
+    }
+
+    /**
+     * @return void
+     */
+    private function setMessageProperties(): void
+    {
+        $settings = $this->getVendorGlobals();
+        $this->md5UserPassword = md5($settings['weno_admin_password'] ?? '');
+        $this->userEmail = $settings['weno_admin_username'] ?? '';
+        $this->encryptionKey = $settings['weno_encryption_key'] ?: 'Missing';
+    }
+
+    /**
+     * @param $key
+     * @return void
+     */
+    public function setNewEncryptionKey($key): void
+    {
+        $gbl = $this->getVendorGlobals();
+        $gbl['weno_encryption_key'] = $key;
+        // save the new key to the database.
+        // save will also set the global to stay current.
+        $this->saveVendorGlobals($gbl);
+        error_log('A new encryption key ' . $key . ' was created and saved: ' . date('Y-d-m H:i:s', time()));
+        $wenoLog = new WenoLogService();
+        $wenoLog->insertWenoLog("new_encryption_key", "saved new key value " . $key);
```

Comment:
also would remove the $key in the output above if this is a sensitive encryption key